pkg/controller/servicebindingrequest/bind_custom_resources.go

```diff
@@ -0,0 +1,104 @@
+package servicebindingrequest
+
+import (
+	"github.com/redhat-developer/service-binding-operator/pkg/apis/apps/v1alpha1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+var (
+	path = map[string][][]string{
+		"ConfigMap": {{"data"}},
+		"Secret":    {{"data"}},
+		"Route":     {{"spec", "host"}},
+		"Service":   {{"spec", "clusterIP"}},
+	}
+)
+
+// BindNonBindableResources struct contains information about operator backed CR and
+// list of expected GVRs to extract information from.
+type BindNonBindableResources struct {
```

Comment:
I don't see anything special in this component related to bindable or non-bindable resources; it collects all the data present in the supported resources and flatten it. Unless I can't foresee properly, this could be renamed to something more meaningful, along the lines of `ResourceOwnerInfo` to convey the message that this is about the owner object and its relationships.

Reply:
@isutton makes sense, I have updated the struct